src/nb_ndarray.cpp

```diff
@@ -335,9 +335,19 @@ ndarray_handle *ndarray_import(PyObject *o, const ndarray_req *req,
                 // c-style strides assumed
                 pass_order = req->req_order == 'C';
             } else {
+                bool size_is_zero = std::any_of(t.shape, t.shape + t.ndim,
```

Comment:
Even if the CI passes, this cannot be merged as is because `std::any_of` is part of the `algorithm` header that we (intentionally) do not import. You will need to write this test manually using plain C/C++.